syncd/NotificationProcessor.cpp

```diff
@@ -397,7 +398,21 @@ void NotificationProcessor::process_on_port_state_change(
          * switch vid.
          */
 
-        oper_stat->port_id = m_translator->translateRidToVid(oper_stat->port_id, SAI_NULL_OBJECT_ID);
+        SWSS_LOG_INFO("Port RID %s state change notification", 
+                sai_serialize_object_id(rid).c_str());
+
+        if (false == m_translator->tryTranslateRidToVid(rid, oper_stat->port_id))
```

Comment:
Just FYI, this is not entirely correct behavior, since after switch create we could receive some notifications with port id's and they maybe not found yet. we actually prevent this from happening by doing "discovery" of all possible objects on the switch after create switch will happen. More likely this scenario will happen when actual port will be removed, similar scenario has been already observed when a fdb flush operation is performed, and then bridge port is removed, same issue happens since fdb notification check will not find existing bridge port.

Anyway i have solution which im working on. to have a set of removed objects id's to see if object was removed.

Reply:
Thanks, good to know.  The discovery will resolve this, but need to enable notification on the switch after discovery?

Reply:
they are already enabled